File: go/Sliding Window/1208. 尽可能使字符串相等.go
package window

func equalSubstring(s string, t string, maxCost int) int {
	left, right, totalCos := 0, 0, 0
	for right < len(s) {
		totalCos += abs(int(s[right] - t[right]))
		if totalCos > maxCost {
			totalCos -= abs(int(s[left] - t[left]))
			left++
		}
		right++
	}
	return right - left
}

func abs(a int) int {
	if a >= 0 {
		return a
	} else {
		return -1 * a
	}
}

File: python/Sliding Window/1208. 尽可能使字符串相等.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        totol_cost = 0
        left,right=0,0
        while right < len(s):
            totol_cost += abs(ord(s[right])-ord(t[right]))
            if totol_cost > maxCost:
                totol_cost -= abs(ord(s[left])-ord(t[left]))
                left += 1
            right+=1
        return right - left

S = Solution()
s = "abcd"
t = "bcdf"
maxCost = 3
print(S.equalSubstring(s,t,maxCost))